salt/states/saltmod.py

```diff
@@ -689,6 +732,137 @@ def runner(name, **kwargs):
     return ret
 
 
+def parallel_runners(name, runners):
+    '''
+    Executes multiple runner modules on the master in parallel.
+
+    .. versionadded:: 2017.x.0 (Nitrogen)
+
+    A separate thread is spawned for each runner. This state is intended to be
+    used with the orchestrate runner in place of the ``saltmod.runner`` state
+    when different tasks should be run in parallel. In general, Salt states are
+    not safe when used concurrently, so ensure that they are used in a safe way
+    (e.g. by only targeting separate minions in parallel tasks).
+
+    name:
+        name identifying this state. The name is provided as part of the
+        output, but not used for anything else.
+
+    runners:
+        list of runners that should be run in parallel. Each element of the
+        list has to be a dictionary. This dictionary's name entry stores the
+        name of the runner function that shall be invoked. The optional kwarg
+        entry stores a dictionary of named arguments that are passed to the
+        runner function.
+
+    .. code-block:: yaml
+
+        parallel-state:
+           saltext.parallel-runner:
+             - runners:
+               - name: state.orchestrate
+                 kwarg:
+                   mods: orchestrate_state_1
+               - name: state.orcestrate
```

Comment:
Slight typo: missing the "h".

Related: the error message for fat-fingered function names is a large traceback. I think the lazy-loader has an easy way for us to check those in advance. @cachedout is that correct?

```python
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/Users/shouse/src/salt/salt/salt/state.py", line 1812, in call
    **cdata['kwargs'])
  File "/Users/shouse/src/salt/salt/salt/loader.py", line 1724, in wrapper
    return f(*args, **kwargs)
  File "/Users/shouse/src/salt/salt/salt/states/saltmod.py", line 784, in parallel_runners
    outputs = _parallel_map(call_runner, runners)
  File "/Users/shouse/src/salt/salt/salt/states/saltmod.py", line 102, in _parallel_map
    six.reraise(exc_type, exc_value, exc_traceback)
  File "/Users/shouse/src/salt/salt/salt/states/saltmod.py", line 90, in run_thread
    outputs[index] = func(inputs[index])
  File "/Users/shouse/src/salt/salt/salt/states/saltmod.py", line 782, in call_runner
    **(runner_config['kwarg']))
  File "/Users/shouse/src/salt/salt/salt/modules/saltutil.py", line 1312, in runner
    full_return=full_return)
  File "/Users/shouse/src/salt/salt/salt/runner.py", line 144, in cmd
    full_return)
  File "/Users/shouse/src/salt/salt/salt/client/mixins.py", line 226, in cmd
    self.functions[fun], arglist, pub_data
  File "/Users/shouse/src/salt/salt/salt/loader.py", line 1095, in __getitem__
    func = super(LazyLoader, self).__getitem__(item)
  File "/Users/shouse/src/salt/salt/salt/utils/lazy.py", line 101, in __getitem__
    raise KeyError(key)
KeyError: 'state.orcestrate'
```